daemon/cmd/kube_proxy_replacement.go

```diff
@@ -361,20 +363,62 @@ func probeKubeProxyReplacementOptions() error {
 	return nil
 }
 
-func probeManagedNeighborSupport() {
+func probeManagedNeighborSupport() error {
```

Comment:
Nit: take `Config` as a parameter and write a unit test for the function?

Reply:
To just test the happy path (no errors) with a privileged test?

Reply:
Yeah pretty much!